community/server/src/main/java/org/neo4j/server/rest/web/TransactionalService.java

```diff
@@ -70,11 +70,11 @@ public Response executeStatementsInNewTransaction( final InputStream input, @Con
         try
         {
             TransactionHandle transactionHandle = facade.newTransactionHandle( uriScheme );
-            return createdResponse( transactionHandle, executeStatements( input, transactionHandle, uriInfo.getRequestUri() ) );
+            return createdResponse( transactionHandle, executeStatements( input, transactionHandle, uriInfo.getBaseUri() ) );
         }
         catch ( TransactionLifecycleException e )
         {
-            return invalidTransaction( e, uriInfo.getRequestUri() );
+            return invalidTransaction( e, uriInfo.getBaseUri() );
```

Comment:
It might be good to have the exception message contain the whole request URI, not just the base URI though. Same for all other usages of `invalidTransaction` as well.